sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,528 @@
+- `publicKey`: `string` hex-encoded
+
+### Method `stx_getAddresses`
+
+`result`
+
+- `addresses`: `{}[]`
+  - `address`: `string` address, Stacks c32-encoded
+  - `publicKey`: `string` hex-encoded
+
+### Method `stx_getAccounts`
+
+> **Comment**: This method is similar to `stx_getAddresses`.
+> It was added to provide better backwards compatibility for applications using Gaia.
+
+`result`
+
+- `accounts`: `{}[]`
+  - `address`: `string` address, Stacks c32-encoded
+  - `publicKey`: `string` hex-encoded
+  - `gaiaHubUrl`: `string` URL
+  - `gaiaAppKey`: `string` hex-encoded
+
+### Method `stx_updateProfile`
+
+`params`
+
+- `profile`: `object` Schema.org Person object
+
+`result`
+
+- `profile`: `object` updated Schema.org Person object
+
+## Listeners
+
+In addition to the request interface, event listeners may be provided via the `.listen` method.
+
+- `provider.listen(event: string, listener: (...args: any[]) => void): Function`
+
+> `provider.listen` should return a "unlisten" function, which can be called to remove the listener.
+
+The event name should be closer to nouns than verbs and doesn't use the `on` prefix from DOM naming conventions.
+
+### Event `accountChange`
+
+`listener: (accounts: {}[]) => void`
+
+> `accounts` as defined above in `stx_getAccounts`.
+> The first account is considered the default account (and may be the only "active" account in a wallet).
+
+## Error Codes
+
+Errors thrown by request methods should match existing JSON-RPC 2.0 error codes.
+This way, the user or an intermediary library can handle them in a standardized way.
+Otherwise, no additional error codes are defined in this SIP.
+
+## JSON Representations
+
+While discussing this SIP, it has become clear that the current Stacks.js representation is confusing to developers.
+Rather, a better solution would be human-readable — for example, rely on string literal enumeration, rather than magic values, which need additional lookups.
+Relying on soley a hex-encoded also poses difficulties when building Stacks enabled web applications.
+
+### Clarity values
+
+Proposed below is an updated interface representation for Clarity primitives for use in Stacks.js and JSON compatible environments.
+
+> **Comment**: For encoding larger than JS `Number` big integers, `string` is used.
+
+`0x00` `int`
+
+```ts
+{
+  type: 'int',
+  value: string // `bigint` compatible
+}
+```
+
+`0x01` `uint`
+
+```ts
+{
+  type: 'uint',
+  value: string // `bigint` compatible
+}
+```
+
+`0x02` `buffer`
+
+```ts
+{
+  type: 'buffer',
+  value: string // hex-encoded string
+}
+```
+
+`0x03` `bool` `true`
+
+```ts
+{
+  type: 'true',
+}
+```
+
+`0x04` `bool` `false`
+
+```ts
+{
+  type: 'false',
+}
+```
+
+`0x05` `address` (aka "standard principal")
+
+```ts
+{
+  type: 'address',
+  value: string // Stacks c32-encoded
+}
+```
+
+`0x06` `contract` (aka "contract principal")
+
+```ts
+{
+  type: 'contract',
+  value: `${string}.${string}` // Stacks c32-encoded, with contract name suffix
```

Comment:
this is in several places, but are backticks intended? in a codeblock, they show up as typed

Reply:
Yes, this is a typescript string type, which essentially constrains string type to a subset, matching the expected criteria (not doing much in this case, e.g. it would likely only fail on a string ending with a period; but it's also nice for developer understanding and type hinting in IDEs — this same type could be used in TS codebases)